docarray/array/storage/weaviate/backend.py

```diff
@@ -0,0 +1,298 @@
+                },
+            ]
+        }
+
+    def _load_or_create_weaviate_schema(self, cls_name: Optional[str] = None):
+        """Create a new weaviate schema for this :class:`DocumentArrayWeaviate` object
+        if not present in weaviate or if ``cls_name`` not provided, else if ``cls_name`` is provided
+        load the object with the given ``cls_name``
+
+        :param cls_name: if provided, load this :class:`DocumentArrayWeaviate` object with
+            the data stored in weaviate. If ``cls_name`` is ``None``, the create a schema in weaviate
+            with a newly generated class name.
+        :return: the schemas of this :class`DocumentArrayWeaviate` object and its meta
+        """
+        if not cls_name:
+            doc_schemas = self._get_schema_by_name(self._get_weaviate_class_name())
+            while self._client.schema.contains(doc_schemas):
+                doc_schemas = self._get_schema_by_name(self._get_weaviate_class_name())
+            self._client.schema.create(doc_schemas)
+            return doc_schemas
+
+        doc_schemas = self._get_schema_by_name(cls_name)
+        if self._client.schema.contains(doc_schemas):
+            return doc_schemas
+
+        self._client.schema.create(doc_schemas)
+        return doc_schemas
+
+    def _update_offset2ids_meta(self):
+        """Update the offset2ids in weaviate the the current local version"""
+        if self._offset2ids_wid is not None and self._client.data_object.exists(
+            self._offset2ids_wid
+        ):
+            self._client.data_object.update(
+                data_object={'_offset2ids': self._offset2ids},
+                class_name=self._meta_name,
+                uuid=self._offset2ids_wid,
+            )
+        else:
+            self._offset2ids_wid = str(uuid.uuid1())
+            self._client.data_object.create(
+                data_object={'_offset2ids': self._offset2ids},
+                class_name=self._meta_name,
+                uuid=self._offset2ids_wid,
+            )
+
+    def _get_offset2ids_meta(self) -> Tuple[List, str]:
+        """Return the offset2ids stored in weaviate along with the name of the schema/class
+        in weaviate that stores meta information of this object
+
+        :return: a tuple with first element as a list of offset2ids and second element
+                 being name of weaviate class/schema of the meta object
+
+        :raises ValueError: error is raised if meta class name is not defined
+        """
+        if not self._meta_name:
+            raise ValueError('meta object is not defined')
+
+        resp = (
+            self._client.query.get(self._meta_name, ['_offset2ids', '_additional {id}'])
+            .do()
+            .get('data', {})
+            .get('Get', {})
+            .get(self._meta_name, [])
+        )
+
+        if not resp:
+            return [], None
+        elif len(resp) == 1:
+            return resp[0]['_offset2ids'], resp[0]['_additional']['id']
```

Comment:
```suggestion
            return resp[0].get('_offset2ids', []), resp[0].get('_additional', {}).get('id')
```

Reply:
Since the length of `resp` is 1, at this point we should expect `_offset2ids` and `_additional__id` to be present in the `dict`, hence I think an error should be raised if these fields are not present like in the current case. More importantly, it would also be easier to debug if we error out early as opposed to going with the suggested approach of accepting invalid values which may have potential of causing issues later down the execution.